### PR TITLE
Subscriber error handling extensive testing

### PR DESCRIPTION
**Pull request by nviennot**
We test that the exception thrown contains what we want
Note that the ugly disconnect alias is to stub it and to put it back

### CODE REVIEW

**Comment by kareemk** [comment]
LGTM

**Comment by michaeldauria** [comment]
:+1:

File: spec/integration/error_spec.rb [hunk 1]
```diff
@@ -22,22 +22,31 @@
...
-      pub = PublisherModel.create
-      pub.update_attributes(:field_1 => 'updated')
-      eventually { @error_handler_called_with.should be_a(Exception) }
-    end
+    context 'on the subscriber side' do
+      before { SubscriberModel.class_eval { validates_format_of :field_1, :without => /death/ } }
+
+      it 'calls the error_handler with an exception' do
+        pub = PublisherModel.create
+        pub.update_attributes(:field_1 => 'death')
```
Comment (yodarjun) [inline spec/integration/error_spec.rb:31]: :finnadie:

**Comment by yodarjun** [comment]
lgtm

**Pull request closed without merging**